activesupport/test/core_ext/module/attribute_accessor_per_thread_test.rb

```diff
@@ -66,19 +82,19 @@ def test_values_should_not_bleed_between_threads
     threads = []
     threads << Thread.new do
       @class.foo = "things"
-      sleep 1
+      Thread.pass
```

Comment:
This change is to try and trigger the race condition, but the downside is that `sleep 1` felt like it was needlessly slowing down our tests. 

by switching the `Thread.current` with `@@class_var` I could see that we were hitting the race condition,  and added the `Thread.pass` to make it more likely to occur, but not sleep for a full second.